stix_shifter_modules/rhacs/stix_transmission/connector.py

```diff
@@ -0,0 +1,393 @@
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+
+    def create_results_connection(self, query, offset, length):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :return: dict
+        """
+        try:
+            return_obj = {}
+            response_dict = {}
+
+            min_range = int(offset)
+            max_range = int(offset) + int(length)
+
+            if max_range > int(self.api_client.get_limit()):
+                max_range = int(self.api_client.get_limit())
+
+            """api call for searching alert based on query"""
+            response_wrapper = self.api_client.get_search_results(query)
+            response_code = response_wrapper.response.status_code
+            response_dict = json.loads(response_wrapper.response.text)
+
+            if response_code == 200:
+                return_obj['success'] = True
+            elif response_code == 401:
+                raise InvalidAuthenticationException(response_dict)
+            total_data = []
+            total_count = 0
+            for firstresult in response_dict['alerts']:
+                # exit if result limit crossed
+                if total_count > max_range:
+                    break
+                if firstresult['id'] is not None:
+                    """second level api call for getting details of particular alert id"""
+                    inner_data = self.api_client.get_inner_results(firstresult['id'])
+                    inner_dict = json.loads(inner_data.response.text)
+                    dt = {}
+                    dt_main = {}
+                    is_process_or_netflow = False
+                    commondataset = {}
+                    commondataset = Connector.get_common_data(inner_dict)
+
+                    if inner_dict.get('deployment') is not None:
+                        if inner_dict['deployment']['containers'] is not None:
+                            dt['Container'] = Connector.flatten_container_data(
+                                inner_dict['deployment']['containers'])
+
+                    if inner_dict.get('violations') is not None and \
+                            len(inner_dict.get('violations')) > 0:
+                        for container in dt['Container']:
+                            # exit if result limit crossed
+                            if total_count > max_range:
+                                break
+                            for item in inner_dict.get('violations'):
+                                if item.get('networkFlowInfo') is not None:
+                                    is_process_or_netflow = True
+                                    if item['time'] is not None:
+                                        commondataset["firstObserved"] = item['time']
```

Comment:
why do you need to modify the time datasource fields to firstObserved at this point?

Reply:
This is required to handle multiple process/netflow violation for a single alert_id. The details has been shared by email yesterday.